packages/backend/src/layers/functionality/statement/get-monobank-statement.service.ts

```diff
@@ -1,131 +1,188 @@
+    return '';
+  } else {
+    return error;
+  }
+}
+function errorStringOrArrayResult(statementPart: string | []) {
+  return isString(statementPart)
+    ? deleteNormalError(statementPart)
+    : statementPart.length;
+}
+// let errorTest = 1; //Uncomment this for error test
 @Injectable()
 export class GetMonobankStatementService {
   constructor(
     private monobankIntegration: MonobankIntegration,
-    private configService: ConfigService,
     private statementStorage: StatementStorage,
+    private configService: ConfigService,
     private featureFlagStorage: FeatureFlagStorage,
     private importAttemptStorage: ImportAttemptStorage,
     private userStorage: UserStorage,
   ) {}
 
-  async getStatement({
-    tokenId,
-    importAttemptId,
-    spaceOwnerEmail,
-  }: IGetStatement) {
-    function formatDate(date: Date) {
-      return ' - ' + format(date, 'dd MMM yyyy HH:mm:ss ') + ' - ';
-    }
-
-    const featureFlags = await this.featureFlagStorage.getFeatureFlags();
+  private async writeLog(
+    importAttemptId: string,
+    message: string,
+    status: ImportAttemptStatusType,
+  ) {
     const importAttempt = await this.importAttemptStorage.getByImportAttemptId(
       importAttemptId,
     );
-
-    const importAttemptInProgress =
-      await this.importAttemptStorage.updateImportAttempt(
-        {
-          status: ImportAttemptStatusType.InProgress,
-          log:
-            importAttempt.log +
-            formatDate(new Date()) +
-            ImportAttemptLogDescription.StartStatementImportExecution +
-            '\n',
-        },
-        importAttemptId,
-      );
-
     await this.importAttemptStorage.updateImportAttempt(
       {
-        status: featureFlags[FeatureName.bypassMonobankRateLimit]
-          ? ImportAttemptStatusType.Failed
-          : ImportAttemptStatusType.Successful,
-        log:
-          importAttemptInProgress.log +
-          formatDate(new Date()) +
-          `${
-            featureFlags[FeatureName.bypassMonobankRateLimit]
-              ? ImportAttemptLogDescription.Failed
-              : ImportAttemptStatusType.Successful
-          }` +
-          '\n',
+        status,
+        log: [importAttempt.log, formatDate(new Date()), message, '\n'].join(
+          '',
+        ),
       },
       importAttemptId,
     );
+  }
 
-    const accountList = await this.statementStorage.getAccountByTokenId(
-      tokenId,
-    );
+  private async fetchDataWithRetryOnError(
+    args: Parameters<MonobankIntegration['getStatement']>[0],
+    iteration = 0,
+  ): Promise<MonobankIntegrationResponse> {
+    const standardDelay = this.configService.get('app.monobankRequestDelay');
+    const result = await this.monobankIntegration.getStatement(args);
+    const featureFlags = await this.featureFlagStorage.getFeatureFlags();
+    //Uncomment this for error test
+    // errorTest++;
+    // if (errorTest >= 5) {
+    //   result.data = 'unknown error' as any;
+    // }
```

Comment:
I believe we shouldn't have it here in the final codebase

Reply:
This code is for the convenience of testing how the import will behave during an unknown error. I deleted it